trade-finance-manager-api/src/graphql/reducers/mappings/deal/mapTotals.js

```diff
@@ -1,25 +1,25 @@
 const { formattedNumber } = require('../../../../utils/number');
-const { calculateNewFacilityValue, calculateAmendmentTotalExposure } = require('../../../helpers/amendment.helpers');
+const { calculateNewFacilityValue, calculateAmendmentTotalExposure, findLatestCompletedAmendment } = require('../../../helpers/amendment.helpers');
 const isValidFacility = require('../../../helpers/isValidFacility.helper');
-const api = require('../../../../v1/api');
 const { CURRENCY } = require('../../../../constants/currency.constant');
 
-const mapTotals = async (facilities) => {
+const mapTotals = (facilities) => {
   const totals = {};
 
   // total value of all facilities
-  const facilitiesValue = await Promise.all(facilities.map(async (facility) => {
+  const facilitiesValue = facilities.map((facility) => {
     if (isValidFacility(facility)) {
-      const { facilitySnapshot, tfm, _id } = facility;
-
-      const latestCompletedAmendmentValue = await api.getLatestCompletedValueAmendment(_id);
+      const { facilitySnapshot, tfm } = facility;
 
       // if latest amendment then returns value of new amendment
-      if (latestCompletedAmendmentValue?.value) {
+      if (facility?.amendments?.length > 0) {
```

Comment:
minor thing, you don't need the comparison with `0`. you can simply use `facility?.amendments?.length`

Reply:
done